packages/apps/plugins/plugin-tree/src/components/Tree/Tree.tsx

```diff
@@ -0,0 +1,321 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import { Square, DotsThreeVertical, X } from '@phosphor-icons/react';
+import React, { type HTMLAttributes, type KeyboardEvent, useEffect, useRef, useState } from 'react';
+
+import { Button, DropdownMenu, Input } from '@dxos/react-ui';
+import { getSize, mx } from '@dxos/react-ui-theme';
+
+import { getNext, getParent, getPrevious, getItems, type Item, getLastDescendent } from './types';
+
+// TODO(burdon): Break/join lines.
+// TODO(burdon): TextObject/MarkdownEditor
+// TODO(burdon): Cut-and-Paste.
+// TODO(burdon): Drag-and-drop.
+// TODO(burdon): Better key nav.
+
+type TreeOptions = Pick<HTMLAttributes<HTMLInputElement>, 'placeholder' | 'spellCheck'> & {
+  checkbox?: boolean;
```

Comment:
The term ‘checkbox’ is a role, but this is extending a collection of input element attributes, so it probably shouldn’t be called this.

There isn’t a good html attribute that encapsulates this, but elsewhere in the repo we use the flag `selectable`.

Reply:
Re `selectable`: in this case though it's really meant to enable something like a task list.